### PR TITLE
Allow start from outer runtime

### PR DESCRIPTION
**Pull request by tnull**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -211,6 +211,19 @@ impl Node {
...
+	}
+
+	/// Starts the necessary background tasks, such as handling events coming from user input,
+	/// LDK/BDK, and the peer-to-peer network.
+	///
+	/// After this returns, the [`Node`] instance can be controlled via the provided API methods in
+	/// a thread-safe manner.
+	///
+	/// FIXME: Document runtime
+	pub fn with_runtime(&self, runtime: Arc<tokio::runtime::Runtime>) -> Result<(), Error> {
```
Comment (jkczyz) [inline src/lib.rs]: Maybe call this `start_with_runtime`?

**Review by jkczyz — APPROVED** [review]

File: src/lib.rs [hunk 2]
```diff
@@ -211,6 +211,19 @@ impl Node {
 	/// After this returns, the [`Node`] instance can be controlled via the provided API methods in
 	/// a thread-safe manner.
 	pub fn start(&self) -> Result<(), Error> {
+		let runtime =
+			Arc::new(tokio::runtime::Builder::new_multi_thread().enable_all().build().unwrap());
+		self.with_runtime(runtime)
+	}
+
+	/// Starts the necessary background tasks, such as handling events coming from user input,
```
Comment (G8XSU) [inline src/lib.rs:219]: super-nit for hack: Not sure if this doc is correct?
did we mean runtime coming from user input?

**Review by G8XSU — APPROVED** [review]

File: src/lib.rs (hunk 1 above)
Comment (tnull) [reply src/lib.rs]: Done

**Comment by tnull** [comment]
Force-pushed with the following changes:

```diff
> git diff-tree -U2  281cb67 c64abf1
diff --git a/src/lib.rs b/src/lib.rs
index 7facb39..41b87bd 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -214,5 +214,5 @@ impl Node {
                let runtime =
                        Arc::new(tokio::runtime::Builder::new_multi_thread().enable_all().build().unwrap());
-               self.with_runtime(runtime)
+               self.start_with_runtime(runtime)
        }

@@ -224,5 +224,5 @@ impl Node {
        ///
        /// FIXME: Document runtime
-       pub fn with_runtime(&self, runtime: Arc<tokio::runtime::Runtime>) -> Result<(), Error> {
+       pub fn start_with_runtime(&self, runtime: Arc<tokio::runtime::Runtime>) -> Result<(), Error> {
                // Acquire a run lock and hold it until we're setup.
                let mut runtime_lock = self.runtime.write().unwrap();
```

Gonna land this

**Pull request merged**